Automattic/WP-Job-Manager
# Introduce stat for jobs impressions.

Fixes #2751

### Changes Proposed in this Pull Request

* Introduce listing impressions on job search.
* Refactors the declarative php and the frontend js side.
* Refactors stats ajax endpoint to accept more/multiple data.

### Testing Instructions

* Open a page that contains the [jobs] shortcode.
* check that on page load, the first 10 listings are logged as impressions.
* Press load more
* Check that impressons are queued up and ajax calls are periodically done as you scroll.


### Release Notes

* 

### New or Updated Hooks and Templates


*


### Deprecated Code

*

### Screenshot / Video




















----

| Plugin build for 868bd05165a48041be7568b66e8bf6c7ab7dcf57 <a href="#"><img width=600></a> |
| ------------------------------------------------------------ |
| 📦 [Download plugin zip](https://wpjobmanager.com/wp-content/uploads/2024/02/wp-job-manager-zip-2771-868bd051.zip)                       |
| ▶️ [Open in playground](https://wpjobmanager.com/playground/?core=2024/02/2771-868bd051)             |
